pkg/snapshot/restorer/restorer.go

```diff
@@ -659,7 +662,14 @@ func applyEventsToEtcd(client *clientv3.Client, events []event) error {
 			return fmt.Errorf("Unexpected event type")
 		}
 	}
-	_, err := client.Txn(ctx).Then(ops...).Commit()
+
+	// Divide the total operations in smaller chunks as mer MaxTxnOps
+	var err error
+	for len(ops)/int(ro.Config.MaxTxnOps) > 0 {
```

Comment:
I find this quite hard to read, but this could be a personal thing, I don't know. Probably, to enhance readability, I would first construct batches of operations, and then iterate again over the batches and apply them. But if this you tested this and it works, go for it. :)